bindings/matic/api.go

```diff
@@ -0,0 +1,117 @@
+package matic
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"net/http"
+)
+
+type BlockIncludedResponse struct {
+	HeaderBlockNumber string `json:"headerBlockNumber"`
+	BlockNumber       string `json:"blockNumber"`
+	Start             string `json:"start"`
+	End               string `json:"end"`
+	Proposer          string `json:"proposer"`
+	Root              string `json:"root"`
+	CreatedAt         string `json:"createdAt"`
+	Message           string `json:"message"`
+}
+
+type BlockIncludedResponseError struct {
+	Error   bool   `json:"error"`
+	Message string `json:"message"`
+}
+
+// API performs requests to various matic APIS.
+type API struct {
+	network Network
+	client  *http.Client
+}
+
+type Network string
+
+const (
+	NetworkMumbai  Network = "MUMBAI"
+	NetworkMainnet Network = "MAINNET"
+)
+
+func NewAPI(network Network, client *http.Client) *API {
+	return &API{
+		network: network,
+		client:  client,
+	}
+}
+
+func (a *API) getBlockIncludedURI(block *big.Int) string {
+	if a.network == NetworkMainnet {
+		return fmt.Sprintf("https://apis.matic.network/api/v1/matic/block-included/%v", block.String())
+	}
+	return fmt.Sprintf("https://apis.matic.network/api/v1/mumbai/block-included/%v", block.String())
+}
+
+var ErrBlockNotIncluded = errors.New("block not yet included")
+
+func (a *API) GetBlockIncludedResponse(block *big.Int) (BlockIncludedResponse, error) {
+	resp, err := a.client.Get(a.getBlockIncludedURI(block))
+	if err != nil {
+		return BlockIncludedResponse{}, err
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return BlockIncludedResponse{}, err
+	}
+
+	res := BlockIncludedResponse{}
+	err = json.Unmarshal(body, &res)
+	if err != nil {
```

Comment:
How is this supposed to work? As far as I know json unmarshal can fail and return an error here on two different cases:
* JSON value not representable by target type
* Syntax error inside json

You also dont use the parsed `res := BlockIncludedResponseError{}` or am I missing something there?

If fields or values are missing, it's not gonna error. Or am I missing something?

Reply:
The matic API returns 2 different responses depending on whether the block was included or not. The structures are different, therefore the flow goes like this:

1) Try parse success response. The response is returned if it was a success.
1) If failed, try to parse the error response. If it succeeded, that means it's not included yet, return ErrBlockNotIncluded. 
1) In case the error response parsing failed, that probably means an API error. Since this is a lib - I want to wrap as many details as I can and return this to the caller as I don't want to inject a logger.